File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsOrgStructureAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

// MARK: - Org structure admin (M14.4)

@Serializable
data class AdminOrgRow(
    val id: String,
    val slug: String,
    val name: String,
    val status: String,
    val dataRegion: String? = null,
    val userCount: Long? = null,
    val courseCount: Long? = null,
    val createdAt: String? = null,
)

@Serializable
data class AdminOrgsListResponse(
    val organizations: List<AdminOrgRow>? = null,
)

@Serializable
data class OrgUnitTreeNode(
    val id: String,
    val name: String,
    val unitType: String,
    val status: String,
    val childCourseCount: Long? = null,
    val children: List<OrgUnitTreeNode>? = null,
)

@Serializable
data class OrgUnitTreeResponse(
    val tree: List<OrgUnitTreeNode>? = null,
)

@Serializable
data class CreateAcademicTermRequest(
    val name: String,
    val termType: String,
    val startDate: String,
    val endDate: String,
)

@Serializable
data class PatchAcademicTermRequest(
    val name: String? = null,
    val termType: String? = null,
    val startDate: String? = null,
    val endDate: String? = null,
    val status: String? = null,
)

@Serializable
data class PatchOrgUnitRequest(
    val name: String? = null,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsOrgStructureAdmin.swift
import Foundation

// MARK: - Org structure admin (M14.4)

struct AdminOrgRow: Decodable, Identifiable, Hashable {
    var id: String
    var slug: String
    var name: String
    var status: String
    var dataRegion: String?
    var userCount: Int64?
    var courseCount: Int64?
    var createdAt: String?
}

struct AdminOrgsListResponse: Decodable {
    var organizations: [AdminOrgRow]?
}

struct OrgUnitTreeNode: Decodable, Identifiable, Hashable {
    var id: String
    var name: String
    var unitType: String
    var status: String
    var childCourseCount: Int64?
    var children: [OrgUnitTreeNode]?
}

struct OrgUnitTreeResponse: Decodable {
    var tree: [OrgUnitTreeNode]?
}

struct CreateAcademicTermRequest: Encodable {
    var name: String
    var termType: String
    var startDate: String
    var endDate: String
}

struct PatchAcademicTermRequest: Encodable {
    var name: String?
    var termType: String?
    var startDate: String?
    var endDate: String?
    var status: String?
}

struct PatchOrgUnitRequest: Encodable {
    var name: String?
}
